### PR TITLE
Allow to configure a docker registry mirror for e2e tests

### PR DESCRIPTION
**Pull request by cclerget**
Continuation of https://github.com/apptainer/singularity/pull/6303

### CODE REVIEW

**Comment by DrDaveD** [comment]
This is a continuation of apptainer/singularity#6303.

Is it ready to review?  Please add a PR description.

**Comment by cclerget** [comment]
@DrDaveD RFR

File: INSTALL.md [hunk 1]
```diff
@@ -222,3 +222,38 @@ For more information on installing/updating/uninstalling the RPM, check out our
 ## Debian Package
 
 Additional information on how to build a Debian package can be found in [dist/debian/DEBIAN_PACKAGE.md](dist/debian/DEBIAN_PACKAGE.md).
+
+## Run E2E tests
+
+The test suite is heavily relying on Docker Hub registry, since the introduction
+of the rate pull limit, developpers can quickly hit the quota limit leading to
```
Comment (DrDaveD) [inline INSTALL.md:229]:
```suggestion
of the rate pull limit, developers can quickly hit the quota limit leading to
```

File: INSTALL.md [hunk 2]
```diff
@@ -222,3 +222,38 @@ For more information on installing/updating/uninstalling the RPM, check out our
...
+2. or you can run a local pull through cache registry and use
+`E2E_DOCKER_MIRROR`/`E2E_DOCKER_MIRROR_INSECURE` environment variables
+
+### Run a local pull through cache registry
+
+The most straightforward way to run it is to run in a terminal:
+
+```sh
+mkdir -p $HOME/.cache/registry
+apptainer run --env REGISTRY_HTTP_ADDR=127.0.0.1:5001 \
```
Comment (DrDaveD) [inline INSTALL.md:247]:
```suggestion
singularity run --env REGISTRY_HTTP_ADDR=127.0.0.1:5001 \
```

Comment (DrDaveD) [reply INSTALL.md:247]: The new name isn't available yet so I think it's better to leave that for a later global change.

**Pull request merged**